### PR TITLE
Move form validation into form.helpers

### PR DESCRIPTION
**Pull request by orangejenny**
This is the last PR on this theme. `app_manager.models` is now under 6000 lines (barely, and doubtless won't stay that way for long).

@emord / @proteusvacuum

### CODE REVIEW

**Review by emord — APPROVED** [review]
No blocking comments

File: corehq/apps/app_manager/helpers/validators.py [hunk 1]
```diff
@@ -543,6 +551,79 @@ def validate_with_raise(self):
...
+
+class IndexedFormBaseValidator(object):
+    def __init__(self, form):
+        self.form = form
+
+    def check_case_properties(self, all_names, subcase_names, case_tag):
+        errors = []
+
+        # reserved_words are hard-coded in three different places!
+        # Here, case-config-ui-*.js, and module_view.html
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/helpers/validators.py]: Indeed, this comment is from at least 2013, though it's less critical than it used to be: https://github.com/dimagi/commcare-hq/commit/ccfc7fbca25e76b091c9489ce98602eea0535bbf#diff-535390eb5c83a4cd3449f82afd1fa9fbL514

Removed.

File: corehq/apps/app_manager/helpers/validators.py [hunk 2]
```diff
@@ -541,3 +541,43 @@ def validate_with_raise(self):
...
             })
         return errors
+
+
+class ShadowFormValidator(object):
+    def __init__(self, form):
+        self.form = form
+
+    @property
+    def timing_context(self):
```
[the remark this answers is not in the document]
Comment (orangejenny) [reply corehq/apps/app_manager/helpers/validators.py]: yes: https://github.com/dimagi/commcare-hq/blob/5410c18f694710147986e735b052e6e65c7ec26c/corehq/util/timer.py#L193

**Comment by orangejenny** [comment]
Killing the build - forgot `ci skip` on the last commit - and merging.

**Pull request merged**